sdks/python/apache_beam/dataframe/frame_base.py

```diff
@@ -218,10 +218,20 @@ def wrapper(*args, **kwargs):
     deferred_arg_indices = []
     deferred_arg_exprs = []
     constant_args = [None] * len(args)
+    from apache_beam.dataframe.frames import _DeferredIndex
     for ix, arg in enumerate(args):
       if isinstance(arg, DeferredBase):
         deferred_arg_indices.append(ix)
         deferred_arg_exprs.append(arg._expr)
+      elif isinstance(arg, _DeferredIndex):
+        deferred_arg_indices.append(ix)
+        deferred_arg_exprs.append(
+            expressions.ComputedExpression(
+                'index_as_series',
+                lambda ix: ix.index.to_series(),  # yapf break
+                [arg._frame._expr],
+                preserves_partition_by=partitionings.Singleton(),
+                requires_partition_by=partitionings.Nothing()))
```

Comment:
Do we want to call `to_series` in all cases? It seems like this expression should provide the index itself to be general purpose.

Reply:
Perhaps. Currently, however, the infrastructure requires anything partition-able to have an index, which indices don't have, so that wouldn't work. I'll drop a TODO.

Reply:
Ah ok, thanks for the explanation